lib/journaled/outbox/metric_emitter.rb

```diff
@@ -2,83 +2,92 @@
 
 module Journaled
   module Outbox
-    # Handles metric emission for the Worker
+    # Handles metric emission for the Worker and Kinesis senders
     #
-    # This class is responsible for collecting and emitting metrics about the outbox queue.
+    # This class provides utility methods for collecting and emitting metrics.
     class MetricEmitter
-      def initialize(worker_id:)
-        @worker_id = worker_id
-      end
+      class << self
+        # Emit batch processing metrics
+        #
+        # @param stats [Hash] Processing statistics with :succeeded, :failed_permanently, :failed_transiently
+        # @param worker_id [String] ID of the worker processing the batch
+        def emit_batch_metrics(stats, worker_id:)
+          total_events = stats[:succeeded] + stats[:failed_permanently] + stats[:failed_transiently]
 
-      # Emit batch processing metrics
-      #
-      # @param stats [Hash] Processing statistics with :succeeded, :failed_permanently, :failed_transiently
-      def emit_batch_metrics(stats)
-        total_events = stats[:succeeded] + stats[:failed_permanently] + stats[:failed_transiently]
+          emit_metric('journaled.worker.batch_process', value: total_events, worker_id:)
+          emit_metric('journaled.worker.batch_sent', value: stats[:succeeded], worker_id:)
+          emit_metric('journaled.worker.batch_failed_permanently', value: stats[:failed_permanently], worker_id:)
+          emit_metric('journaled.worker.batch_failed_transiently', value: stats[:failed_transiently], worker_id:)
```

Comment:
```suggestion
          emit_metric('journaled.worker.batch.processed', value: total_events, worker_id:)
          emit_metric('journaled.worker.batch.sent', value: stats[:succeeded], worker_id:)
          emit_metric('journaled.worker.batch.failed_permanently', value: stats[:failed_permanently], worker_id:)
          emit_metric('journaled.worker.batch.failed_transiently', value: stats[:failed_transiently], worker_id:)
```

🤷 nothing says we can't have more than 3 `.` in a stat name -- but mainly i renamed `process` to `processed`.